src/otx/core/model/visual_prompting.py

```diff
@@ -577,9 +582,14 @@ def _customize_outputs(
         """Customize OTX output batch data entity if needed for model."""
         masks: list[tv_tensors.Mask] = []
         scores: list[torch.Tensor] = []
-        for output in outputs:
-            masks.append(torch.as_tensor(output["hard_prediction"], device=self.device))
-            scores.append(torch.as_tensor(output["scores"], device=self.device))
+        for image_output in outputs:
+            masks.extend(
+                [
+                    torch.as_tensor(hard_prediction, device=self.device)
+                    for hard_prediction in image_output.hard_predictions
+                ],
+            )
+            scores.extend([torch.as_tensor(score, device=self.device) for score in image_output.scores])
```

Comment:
It seems that `len(masks)` and `len(scores)` aren't matched with `len(outputs)`, doesn't it?

Reply:
Sorry for that because it seems implementation bug.
Currently, batch size for test is 1, so it doesn't matter to use flattened outputs.
However, for multi batch testing, entity has batch-wise components (masks and scores), not flattened one.
I'll fix it quickly, please reflect it.

Reply:
https://github.com/openvinotoolkit/training_extensions/pull/3695

Reply:
Let's fix after this PR is merged